packages/synapse-react-client/src/synapse-queries/KeyFactory.ts

```diff
@@ -44,17 +44,18 @@ const entityQueryKeyObjects = {
   /* Query key for all entities */
   all: { objectType: 'entity' },
   // Data for an entity
-  entity: (id: string) => ({ ...entityQueryKeyObjects.all, id: id }),
-  // Data for a particular version of an entity
-  version: (id: string, versionNumber?: string | number) => ({
-    ...entityQueryKeyObjects.entity(id),
-    versionNumber,
-  }),
+  entity: (id: string, versionNumber?: string | number) => {
+    const key: Record<string, unknown> = {
+      ...entityQueryKeyObjects.all,
+      id: id,
+    }
+    if (versionNumber) {
+      key['version'] = versionNumber
+    }
+    return key
+  },
```

Comment:
Change the query key so our query keys look like 
```
[ { objectType: 'entity', id: 'syn123' } ]
[ { objectType: 'entity', id: 'syn123', version: 2 } ]

[ { objectType: 'entity', id: 'syn123' }, 'activity' ]
[ { objectType: 'entity', id: 'syn123', version: 2 }, 'activity' ]
```

I verified that the invalidations will work as expected. For example, calling `queryClient.invalidateQueries` with the query key  `[ { objectType: 'entity', id: 'syn123' } ]` will invalidate all queries related to syn123, including those that include a version number.